pkg/imgpkg/registry/registry.go

```diff
@@ -222,25 +221,13 @@ func newHTTPTransport(opts Opts) (*http.Transport, error) {
 		}
 	}
 
-	// Copied from https://github.com/golang/go/blob/release-branch.go1.12/src/net/http/transport.go#L42-L53
-	// We want to use the DefaultTransport but change its TLSClientConfig. There
-	// isn't a clean way to do this yet: https://github.com/golang/go/issues/26013
-	return &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		DialContext: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-			DualStack: true,
-		}).DialContext,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ResponseHeaderTimeout: 10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-		// Use the cert pool with k8s cert bundle appended.
-		TLSClientConfig: &tls.Config{
-			RootCAs:            pool,
-			InsecureSkipVerify: (opts.VerifyCerts == false),
-		},
-	}, nil
+	clonedDefaultTransport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
the only downside i see here is that if something changes default transport this code will be affected. do we know if anything changing default transport anywhere?

Reply:
the upside is that as we upgrade the http package, we get any changes / updates made to the DefaultTransport. (we avoid drifting from DefaultTransport)

i.e. since inlining the DefaultTransport, we drifted in 2 fields:
- `ForceAttemptHTTP2` setting to true 
- `DialContext.DualStack` has been deprecated 

sounds like a trade-off to me. While the risk you outline is true, currently nothing is mutating the `DefaultTransport` in our codebase or any of our libraries.